src/sw.ts

```diff
@@ -407,8 +423,29 @@ async function fetchHandler ({ path, request }: FetchHandlerArg): Promise<Respon
    * * https://bugs.chromium.org/p/chromium/issues/detail?id=823697
    * * https://bugzilla.mozilla.org/show_bug.cgi?id=1394102
    */
-  // 5 minute timeout
-  const signal = AbortSignal.timeout(5 * 60 * 1000)
+  const abortController = new AbortController()
+  const signal = abortController.signal
+  const abortFn = (event: Pick<AbortSignalEventMap['abort'], 'type'>): void => {
+    clearTimeout(signalAbortTimeout)
```

Comment:
I think you'll need to move `signalAbortTimeout` above this function, since `const` statements are not hoisted like `vars` and might not be available in the closure when you define `abortFn`

Reply:
const is "hoisted" into it's scoped closure (i.e. available in the scopes lookup table), which `abortFn` is also a part of

short example:

```
(() => {
    const abortFn = () => { console.log(varDefinedAfterFn) }
    const varDefinedAfterFn = 'hello world'
    abortFn() // hello world
})()
```

If they were in different scopes.. it wouldn't:

```
(() => {
    const abortFn = () => { console.log(varDefinedAfterFn) }
    {const varDefinedAfterFn = 'hello world'}
    abortFn() // Uncaught ReferenceError: varDefinedAfterFn is not defined
})()
```

Reply:
Thanks for clarifying that.